radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/data/PreparedTransaction.java

```diff
@@ -77,19 +81,22 @@
 	private final byte[] blob;
 	private final byte[] hashToSign;
 	private final UInt256 fee;
+	private final List<JSONObject> notifications;
 
-	private PreparedTransaction(byte[] blob, byte[] hashToSign, UInt256 fee) {
+	private PreparedTransaction(byte[] blob, byte[] hashToSign, UInt256 fee, List<JSONObject> notifications) {
 		this.blob = blob;
 		this.hashToSign = hashToSign;
 		this.fee = fee;
+		this.notifications = notifications;
 	}
 
-	public static PreparedTransaction create(byte[] blob, byte[] hashToSign, UInt256 fee) {
+	public static PreparedTransaction create(byte[] blob, byte[] hashToSign, UInt256 fee, List<JSONObject> notifications) {
 		requireNonNull(blob);
 		requireNonNull(hashToSign);
 		requireNonNull(fee);
+		requireNonNull(notifications);
```

Comment:
Can we add a message as a second parameter like "blob must not be null"?

Reply:
This method is intended for use by Jackson, not by programmer. And in this scenario this check is even unnecessary as Jackson provides much more detailed error description when it happens.

Reply:
If it is not necessary then I'd remove them.

Reply:
It's better to be safe than sorry. Someone might try to create instance manually and then check will be useful.